test/integration/test_konnect_extension.go

```diff
@@ -100,174 +99,243 @@ func TestKonnectExtension(t *testing.T) {
-							Name: authCfg.Name,
-						},
-					}),
-					deploy.WithKonnectIDControlPlaneRef(cp),
-					setKonnectExtensionDPCertSecretRef(t, secretCert),
-				)
-				t.Cleanup(deleteObjectAndWaitForDeletionFn(t, keWithKonnectIDCPRef.DeepCopy()))
-
-				params := KonnectExtensionTestBodyParams{
-					konnectControlPlane: cp,
-					konnectExtension:    keWithKonnectIDCPRef,
-					secret:              secretCert,
-					client:              clientNamespaced,
-					authConfigName:      authCfg.Name,
-					namespace:           ns.Name,
-				}
-				konnectExtensionTestBody(t, params)
+			// run the KonnectExtension test cases.
+			KonnectExtensionTestCases(t, KonnectExtensionTestCaseParams{
+				konnectControlPlane: cp,
+				service:             service,
+				client:              clientNamespaced,
+				namespace:           ns.Name,
+				authConfigName:      authCfg.Name,
 			})
+		})
 
-			t.Run("automatic secret provisioning", func(t *testing.T) {
-				keWithKonnectIDCPRef := deploy.KonnectExtension(
-					t, ctx, clientNamespaced,
-					deploy.WithKonnectConfiguration[*konnectv1alpha1.KonnectExtension](konnectv1alpha1.KonnectConfiguration{
-						APIAuthConfigurationRef: konnectv1alpha1.KonnectAPIAuthConfigurationRef{
-							Name: authCfg.Name,
-						},
-					}),
-					deploy.WithKonnectIDControlPlaneRef(cp),
-				)
-				t.Cleanup(deleteObjectAndWaitForDeletionFn(t, keWithKonnectIDCPRef.DeepCopy()))
-				params := KonnectExtensionTestBodyParams{
-					konnectControlPlane: cp,
-					konnectExtension:    keWithKonnectIDCPRef,
-					secret:              nil, // automatic provisioning
-					client:              clientNamespaced,
-					authConfigName:      authCfg.Name,
-					namespace:           ns.Name,
-				}
-				konnectExtensionTestBody(t, params)
+		t.Run("Mirror ControlPlane", func(t *testing.T) {
+			// Create a Mirror Konnect control plane for the KonnectExtension to attach to.
+			mirrorCP := deploy.KonnectGatewayControlPlane(t, GetCtx(), clientNamespaced, authCfg,
+				deploy.WithTestIDLabel(testID),
+				deploy.WithMirrorSource(cp.GetKonnectID()),
+			)
+			t.Cleanup(deleteObjectAndWaitForDeletionFn(t, mirrorCP.DeepCopy()))
+
+			t.Logf("Waiting for Konnect ID to be assigned to ControlPlane %s/%s", mirrorCP.Namespace, mirrorCP.Name)
+			require.EventuallyWithT(t, func(t *assert.CollectT) {
+				err := GetClients().MgrClient.Get(GetCtx(), k8stypes.NamespacedName{Name: mirrorCP.Name, Namespace: mirrorCP.Namespace}, mirrorCP)
+				require.NoError(t, err)
+				assertKonnectEntityProgrammed(t, mirrorCP)
+			}, testutils.ObjectUpdateTimeout, testutils.ObjectUpdateTick)
+			require.EventuallyWithT(t, func(t *assert.CollectT) {
+				err := GetClients().MgrClient.Get(GetCtx(), k8stypes.NamespacedName{Name: mirrorCP.Name, Namespace: mirrorCP.Namespace}, mirrorCP)
+				require.NoError(t, err)
+				assertKonnectGatewayControlPlaneMirrored(t, mirrorCP)
+			}, testutils.ObjectUpdateTimeout, testutils.ObjectUpdateTick)
```

Comment:
For this (and to substitute `assertKonnectGatewayControlPlaneMirrored`) we can use the [`Predicates` type](https://github.com/Kong/gateway-operator/blob/bbd79606a2dfb9f87722bb5bb4065d0bf9cd82c7/pkg/utils/test/predicates_conditions.go#L15) just like it's done in https://github.com/Kong/gateway-operator/blob/bbd79606a2dfb9f87722bb5bb4065d0bf9cd82c7/test/integration/test_controlplane.go#L498-L511

WDYT?